### PR TITLE
Provide defaulting for textScaleFactor when passing to dart:ui

### LINKED ISSUE #66196: RichText textScaleFactor is ignored if the child TextSpan doesn't contain a TextStyle
## Steps to Reproduce
1. Run this dartpad [code sample](https://dartpad.dartlang.org/b65e7edd460d38507ff39162021030be)

**Expected results:** 
[RichText](https://api.flutter.dev/flutter/widgets/RichText-class.html) [textScaleFactor property](https://api.flutter.dev/flutter/widgets/RichText/textScaleFactor.html) should apply to his child [TextSpan](https://api.flutter.dev/flutter/painting/TextSpan-class.html)

**Actual results:** 
[RichText](https://api.flutter.dev/flutter/widgets/RichText-class.html) [textScaleFactor property](https://api.flutter.dev/flutter/widgets/RichText/textScaleFactor.html) doesn't apply to his child [TextSpan](https://api.flutter.dev/flutter/painting/TextSpan-class.html) unless [TextStyle](https://api.flutter.dev/flutter/painting/TextStyle-class.html) is declared
(try un-commenting the code as described in the sample)

I have no relevant logs to provide

<details>
  <summary>doctor</summary>

```console
[✓] Flutter (Channel master, 1.22.0-10.0.pre.269, on Linux, locale en_US.UTF-8)
    • Flutter version 1.22.0-10.0.pre.269 at
      /home/francesco/development/flutter_master
    • Framework revision 51fa704642 (71 minutes ago), 2020-09-19 11:27:04 -0700
    • Engine revision 4b8477d115
    • Dart version 2.10.0 (build 2.10.0-142.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /usr/lib/Android
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /usr/lib/Android
    • ANDROID_SDK_ROOT = /usr/lib/Android
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_265-8u265-b01-0ubuntu2~18.04-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from
      https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup
      for detailed instructions).

[✓] Connected device (3 available)
    • Linux (desktop)  • linux      • linux-x64      • Linux
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome
      85.0.4183.102

! Doctor found issues in 1 category.
```

</details>

please consider this [PR](https://github.com/flutter/flutter/pull/66197) to fix the issue

### PR DESCRIPTION
**Pull request by GaryQian**
## Description

`textScaleFactor` depends on an existing fontSize to be multiplied with in order to be passed into dart:ui. TextSpans using RichText do not inherit styles from parents, and thus, fail to pass `textScaleFactor` when style is null.

Here, we provide the default font size as a base font to be multiplied by. This does not change behavior other than fix the bug, since the default font size would have been applied at the engine level anyways.

## Related Issues

Fixes #66196

and https://github.com/flutter/flutter/pull/66197

### CODE REVIEW

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 1]
```diff
@@ -16,6 +16,11 @@ import 'text_span.dart';
 
 export 'package:flutter/services.dart' show TextRange, TextSelection;
 
+// The default font size if none is specified. This should be kept in
+// sync with the default values in text_style.dart, as well as the
+// defaults set in the engine (eg, LibTxt's text_style.h, paragraph_style.h).
```
Comment (GaryQian) [inline packages/flutter/lib/src/painting/text_painter.dart:21]: These are not formal API docs, so I included a reference to LibTxt, I can remove if people feel it is best not to link to specific engine implementation that is slated to be swapped out.

**Review by Piinks — APPROVED** [review]
Nice! This LGTM!

File: packages/flutter/lib/src/painting/text_painter.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/painting/text_painter.dart:21]: I trust you to update this comment once libtxt is swapped out :)

Comment (justinmc) [reply packages/flutter/lib/src/painting/text_painter.dart:21]: If you feel like an overachiever though, you could consider moving _kDefaultFontSize to a shared constants file in painting/constants.dart.  We have a few files like that elsewhere but none in painting yet.

**Review by justinmc — APPROVED** [review]
LGTM 👍

**Pull request merged**